File: files/CH-154.py
import pandas as pd

path = "CH-154 Custom Index Column.xlsx"
input = pd.read_excel(path, usecols="B:C", skiprows=1, nrows=11)
test = pd.read_excel(path, usecols="E:G", skiprows=1, nrows=11)

input['Date'] = pd.to_datetime(input['Date'])
input['index'] = (input['Date'].diff() != pd.Timedelta(days=1)).cumsum().astype('int64')

print(all(input['index'] == test['index'])) # True

File: files/100-199/154/CH-154.R
library(tidyverse)
library(readxl)

path = "files/CH-154 Custom Index Column.xlsx"
input = read_excel(path, range = "B2:C13")
test  = read_excel(path, range = "E2:G13")

result = input %>%
  mutate(index = cumsum(c(0, diff(Date)) != 1)) 

all.equal(result$index, test$index)
#> [1] TRUE          